Wangscape/utils.h

```diff
@@ -0,0 +1,64 @@
+#pragma once
+#include <type_traits>
+#include <stdexcept>
+
+namespace
+{
+
+template<typename I>
+I _pow(I base, I exp)
```

Comment:
For me type promotions are enough and a bit wider range of `unsigned long long` positive values irrelevant, but if you think it's important, keep this template :).